src/contentRepo.ts

```diff
@@ -0,0 +1,37 @@
+import { MatrixClient } from "matrix-bot-sdk";
+
+/**
+ * Get the HTTP URL for an MXC URI.
+ * @param {string} baseUrl The base homeserver url which has a content repo.
+ * @param {string} mxc The mxc:// URI.
+ * @param {Number} width The desired width of the thumbnail.
+ * @param {Number} height The desired height of the thumbnail.
+ * @param resizeMethod The thumbnail resize method to use, either
+ * "crop" or "scale".
+ * @param allowDirectLinks If true, return any non-mxc URLs
+ * directly. Fetching such URLs will leak information about the user to
+ * anyone they share a room with. If false, will return the emptry string
+ * for such URLs.
+ * @return The complete URL to the content. May be empty string if mxc is not a string.
+ */
+function getHttpUriForMxc(baseUrl: string, mxc: string, width?: number, height?: number,
```

Comment:
Is this borrowed from js/bot-sdk, or written anew? If borrowed, I'd permalink the original source in a comment in case it needs longer than we're planning it to and needs some upstream changes backported.

Reply:
It's a weird one, the signature is borrowed but the implementation is written anew. What's best do you think?